cornucopia/geometric.py

```diff
@@ -1230,16 +1229,16 @@ def apply(self, x):
             y = warps.apply_flow(x[None], coord[None], has_identity=True,
                                  padding_mode=self.bound)[0]
 
-            if y.is_floating_point():
-                # not a label map -> apply PSF
-                kernel = torch.full(
-                    [nb_repeat, 1],
-                    1 / math.sqrt(nb_repeat),
-                    dtype=y.dtype, device=y.device
-                )
-                y = y.movedim(zindex, -1).unfold(-1, nb_repeat, nb_repeat)
-                y = y.matmul(kernel).matmul(kernel.t())  # PSF + replicate
-                y = y.flatten(-2).movedim(-1, zindex)    # [C, *oshape]
+            # if y.is_floating_point():
```

Comment:
Why did you remove this test. Do you sometimes apply the PSF to label maps? I am surprised the code within the if block even runs when `y` is not float (the matmul should fail)